topics/dev/tutorials/tool-from-notebook/tutorial.md

```diff
@@ -0,0 +1,287 @@
+---
+
+layout: tutorial_hands_on
+title: "nb2workflow: Generating Galaxy Tools From Jupyter Notebooks"
+key_points:
+  - The nb2galaxy module of the nb2workflow library is an automated Galaxy tool generator for scientists and developers who routinely write Jupyter Notebooks.
+  - nb2workflow is a Python package that can be easily installed via pip.
+  - Once a Jupyter Notebook is functional, only a clear separation and definition of the inputs and outputs - along with appropriate annotations and tags - is needed for conversion.
+  - The conversion module accepts additional files (e.g., help text, citations, environment specifications) to ensure the generated tool passes planemo lint checks.
+
+objectives:
+ - Learn why you might want to use the nb2galaxy.
+ - Install the package and explore the provided example notebooks.
+ - Modify the examples and regenerate the tools to check the effects.
+ - Create new Galaxy tools from your own Jupyter Notebooks.
+
+questions:
+ - How can new-to-Galaxy developers convert functioning Jupyter Notebooks into Galaxy tools?
+ - How are the inputs and outputs defined?
+ - How are the tool dependencies provided?
+
+time_estimation: 1H
+subtopic: tooldev
+
+requirements:
+  - type: "internal"
+    topic_name: introduction
+    tutorials:
+      - galaxy-intro-short
+      - galaxy-intro-101-everyone
+  - type: "internal"
+    topic_name: dev
+    tutorials:
+      - tool-integration
+
+
+contributions:
+  authorship:
+    - dsavchenko
+    - Andrei-EPFL
+  editing:
+    - Andrei-EPFL
+
+---
+
+With the advance of python and Jupyter Notebooks, many astronomers have started using them for their data analysis pipelines. In response, members of the astronomy and data science communities have developed `nb2galaxy` -- a module of the `nb2workflow` library -- that allows for a quick and easy conversion of Jupyter Notebooks into Galaxy tools. While originally motivated by astronomical use cases, the tool is broadly applicable. This tutorial is therefore aimed at developers and researchers across disciplines who are interested in Galaxy.
+
+
+> <agenda-title></agenda-title>
+>
+> 1. TOC
+> {:toc}
+>
+{: .agenda}
+
+# Convert an example Jupyter notebook into a Galaxy tool
+
+This tutorial starts with the setup of the required enviornment and a very basic Jupyter Notebook. Afterwards, each section introduces one or more concepts related to the conversion. The tutorial ends with a more complex example of inputs and outputs.
+It is important to mention that the end result of this tutorial is a folder with a `.xml` file that describes the tool and other required files for the tool to function. In order for the tool to be on the UseGalaxy platform, one needs to publish the resulting folder on [toolshed](https://toolshed.g2.bx.psu.edu/) and ...
```

Comment:
TODO finish the sentence
UseGalaxy --> usegalaxy.eu etc (different instances may have different procedures).

Do we need this info here? It's general knowledge in Galaxy world.